lib/openhab/core/configuration.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require "forwardable"
+require_relative "emulate_hash"
+
+module OpenHAB
+  module Core
+    java_import org.openhab.core.config.core.Configuration
+
+    #
+    # {Configuration} represents openHAB's {org.openhab.core.config.core.Configuration} data
+    # with the full interface of {::Hash}.
+    #
+    # All keys are converted to strings.
+    #
+    class Configuration
+      include EmulateHash
+      extend Forwardable
+
+      def_delegator :get_properties, :to_hash
+      def_delegator :to_h, :inspect
+
+      # @!visibility private
+      alias_method :to_s, :inspect
+
+      # @!visibility private
+      def fetch(key, *default_value, &block)
```

Comment:
this can be mostly delegated to java.util.Map#fetch against get_properties, which then becomes the same implementation as Hash, and thus should be in EmulateHash

Reply:
Configuration#getProperties creates an unmodifiableMap out of its properties hashmap.
Calling Configuration#get bypasses that, so I think this should stay, unless we don't care about the additional overheads of unmodifiableMap.

Reply:
OK, I've decided to follow your advice and put `fetch` inside EmulateHash, delegating it to `to_map`, but override `[]` to call `get`.